pkg/controller/podautoscaler/horizontal.go

```diff
@@ -207,14 +207,18 @@ func (a *HorizontalController) processNextWorkItem() bool {
 	}
 	defer a.queue.Done(key)
 
-	err := a.reconcileKey(key.(string))
-	if err == nil {
-		// don't "forget" here because we want to only process a given HPA once per resync interval
-		return true
+	deleted, err := a.reconcileKey(key.(string))
```

Comment:
you should still have a comment as to why we're not forgetting the key, since this works significantly different from normal controllers.